osm2streets/src/output.rs

```diff
@@ -87,54 +72,179 @@ impl StreetNetwork {
-                            ));
+                        (Some(TrafficClass::Motor), Some(TrafficClass::Bicycle))
+                        | (Some(TrafficClass::Bicycle), Some(TrafficClass::Motor)) => {
+                            // AU specifies the use of an "edge line" in this case...
+                            LongitudinalLine::lane(false, false)
+                        }
+                        (Some(TrafficClass::Motor), _) | (_, Some(TrafficClass::Motor)) => {
+                            LongitudinalLine::edge()
                         }
+                        (Some(TrafficClass::Bicycle), Some(TrafficClass::Bicycle)) => {
+                            if pair[0].dir != pair[1].dir {
+                                LongitudinalLine::dividing(guess_overtaking, guess_overtaking)
+                            } else {
+                                LongitudinalLine::lane(true, true)
+                            }
+                        }
+                        (Some(TrafficClass::Bicycle), _) | (_, Some(TrafficClass::Bicycle)) => {
+                            LongitudinalLine::edge()
+                        }
+                        _ => {
+                            continue;
+                        }
+                    };
+                    markings.push(RoadMarking::longitudinal(
+                        separation,
+                        kind,
+                        [pair[0].lt, pair[1].lt],
+                    ));
+                }
+            }
+            // Add the right road edge.
+            if let Some(last_lane) = road.lane_specs_ltr.last() {
+                if matches!(
+                    last_lane.lt.traffic_class(),
+                    Some(TrafficClass::Bicycle) | Some(TrafficClass::Motor)
+                ) {
+                    if let Ok(edge_line) = lane_centers
+                        .last()
+                        .expect("lane_centers to have the same length as lane_specs_ltr")
+                        .shift_right(last_lane.width / 2.0)
+                    {
+                        markings.push(RoadMarking::longitudinal(
+                            edge_line,
+                            LongitudinalLine::edge(),
+                            [last_lane.lt, LaneType::Buffer(BufferType::Verge)],
+                        ));
                     }
-                    continue;
                 }
+            }
+
+            // Add stop and yield lines.
+            // While stop lines are measured against the reference line, we need the reference line
+            // offset. Not bothering to support complicated positioning yet.
+            let ref_offset = road.left_edge_offset_of(
+                if let Placement::Consistent(p) = road.reference_line_placement {
+                    p
+                } else {
+                    RoadPosition::Center
+                },
+                self.config.driving_side,
+            );
```

Comment:
So, we can still make use of the reference line at the end of the process. The code below produces stop lines that will always cross the origination OSM node (I think; as long as placement is `Consistent`). But this shows the benefit of simplifying to a consistently placed `center_line` early on.